lighthouse-core/lib/network-recorder.js

```diff
@@ -99,6 +99,18 @@ class NetworkRecorder extends EventEmitter {
     return !!(isQUIC && receivedHeaders && record.endTime);
   }
 
+  /**
+   * frame root network requests don't always "finish" even when they're done loading data, use responseReceived instead
+   * @see https://github.com/GoogleChrome/lighthouse/issues/6067#issuecomment-423211201
+   * @param {LH.Artifacts.NetworkRequest} record
+   * @return {boolean}
+   */
+  static _isFrameRootRequestAndFinished(record) {
+    const isFrameRootRequest = record.url === record.documentURL;
```

Comment:
do we have to worry about query strings or anything else throwing this check off? (are these fields canonicalized differently, I guess)

Reply:
From what I could tell this is always the full query string-included URL. There were lots of samples on the verge that had misc query string params.